File: week1/Ariling/boj-10825-국영수/main.py
import sys
input = sys.stdin.readline

N = int(input())
arr = []
for _ in range(N):
    infoes = input().split()
    arr.append((infoes[0], int(infoes[1]), int(infoes[2]), int(infoes[3])))
arr = sorted(arr, key=lambda x: (-x[1], x[2], -x[3], x[0]))

for i in range(len(arr)):
    print(arr[i][0])

File: week1/Ariling/boj-10825-국영수/main.js
// 문제 풀이 함수
function solution(input) {
  input.sort(
    (a, b) =>
      Number(b[1]) - Number(a[1]) ||
      Number(a[2]) - Number(b[2]) ||
      Number(b[3]) - Number(a[3]) ||
      (a[0] > b[0] ? 1 : -1)
  );
  for (let x = 0; x < input.length; x++) {
    console.log(input[x][0]);
  }
}

const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let n;
const arr = [];

rl.on("line", (line) => {
  if (!n) {
    n = parseInt(line);
  } else {
    arr.push(line.split(" "));
    if (arr.length === n) {
      rl.close();
    }
  }
}).on("close", () => {
  solution(arr);
  process.exit();
});
